language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -60,6 +60,25 @@ fn consume_token<'input>(
     Ok(())
 }
 
+fn consume_end_of_type_args_list<'input>(
+    tokens: &mut Lexer<'input>,
+) -> Result<(), ParseError<usize, anyhow::Error>> {
+    match tokens.peek() {
+        Tok::Greater => {
+            tokens.advance()?;
+        }
+        Tok::GreaterGreater => {
+            tokens.replace_tok(Tok::Greater, 1)?;
```

Comment:
You're using a positive delta when replacing ">>" with ">" but it should be a "-1" delta to move the end location back by one character, right?